### PR TITLE
jdt-language-server: move to `pkgs/by-name`, set `passthru.updateScript`

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

`jdt-language-server`: set`passthru.updateScript`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by NotAShelf — APPROVED** [review]

**Comment by matt-snider** [comment]
Hey @eclairevoyant, there is another PR that @huantianad opened - #279285

I think this is probably the better approach since it will reduce maintenance in the long term by using the official startup script.

What are your thoughts? Feel free to comment on the other PR as well.

**Comment by eclairevoyant** [comment]
The main point of my PR is to add an update script, I can just cut out the other changes 👍

**Comment by huantianad** [comment]
I can pull in the update script to my PR if wanted, but it's probably better to keep it separate. Up to you!

**Comment by eclairevoyant** [comment]
I think it makes sense as separate PRs, they're independent and combining would make it harder to review imo

**Comment by matt-snider** [comment]
Nice! That makes sense to me. It should be merged after the other PR since it depends on it (due to "hash")

**Review by matt-snider — APPROVED** [review]
This looks good to me and #279285 is merged now - nice work @eclairevoyant!

File: pkgs/development/tools/language-servers/jdt-language-server/update.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash --pure -p curl cacert libxml2 yq nix jq
+
+set -o errexit
+set -o nounset
```
Comment (ambroisie) [inline pkgs/development/tools/language-servers/jdt-language-server/update.sh:5]: Usually just written `set -euo pipefail` (which also enables `pipefail` :-) ).

```suggestion
set -euo pipefail
```

**Review by ambroisie — CHANGES_REQUESTED** [review]
Tiny nit.

**Review by ambroisie — APPROVED** [review]

**Pull request merged**